src/superface/config/prepare-ast.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  assertMapDocumentNode,
+  assertProfileDocumentNode,
+  MapDocumentNode,
+  NormalizedSuperJsonDocument,
+  ProfileDocumentNode,
+} from '@superfaceai/ast';
+import { IFileSystem, Profile } from '@superfaceai/one-sdk';
+import { parseMap, parseProfile, Source } from '@superfaceai/parser';
+
+import { MapUndefinedError, ProfileUndefinedError } from '../../common/errors';
+
+export async function getProfileAst(
+  profile: Profile | string,
+  superJson: NormalizedSuperJsonDocument,
+  fileSystem: IFileSystem
+): Promise<ProfileDocumentNode> {
+  if (profile instanceof Profile) {
+    return profile.ast;
+  } else {
+    const profileSettings = superJson.profiles[profile];
+
+    if (!profileSettings || !('file' in profileSettings)) {
+      throw new ProfileUndefinedError(profile);
+    }
+
+    // TODO: resolve path
+    const profilePath = profileSettings.file;
+    const content = await fileSystem.readFile(profilePath);
+
+    if (content.isErr()) {
+      throw content.error;
+    }
+
+    // TODO: make parser optional
```

Comment:
Remove TODO

Reply:
I added this, I can add dynamic import and call the fallback without parser if import fails. The question is, what will be fallback. 😄

Reply:
Fall should be error saying that you need to compile and there also should be extension check. But I am ok with keeping parser in deps.